### PR TITLE
Features/guyber/run signal from emulator

### PR DESCRIPTION
**Pull request by bermanguy**
1. Move ServiceAnalysisFactory (and all its dependencies) to Shared in order to run signal in emulator
2. Take cadences from manifest and show it in a comboboxe
3. Small layout changes
new

### CODE REVIEW

File: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml [hunk 1]
```diff
@@ -14,29 +14,28 @@
...
             <ColumnDefinition Width="*"/>
         </Grid.ColumnDefinitions>
 
         <Border Grid.Column="0" BorderBrush="#FFA5A5A5" BorderThickness="0,0,0.5,0" Padding="20,20,20,0">
             <StackPanel Orientation="Vertical">
-                <TextBlock Style="{StaticResource SectionHeader}" Text="EMULATION SETTINGS - My Signal 1"/>
-                <TextBlock Style="{StaticResource ControlText}" Text="Start date" Margin="0,0,0,10"/>
-                <DatePicker Margin="0,0,0,20" HorizontalAlignment="Left"/>
+                <TextBlock Style="{StaticResource SectionHeader}" Text="SETTINGS"/>
+                <TextBlock Style="{StaticResource SignalNameTitle}" Text="{Binding SignalName}" />
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml]: why did you split it to two lines?

**Comment by EranGon** [comment]
>                 <Button Style="{StaticResource RoundedButton}" HorizontalAlignment="Left" Content="Start analysis" Command="{Binding RunSignalCommand}" />

this should be disabled while the emulation is running

---
Refers to: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml:55 in c1e6417. [](commit_id = c1e6417df32933139bc81cdf5dba5f72e21b4f7b, deletion_comment = False)

File: src/SDK/SmartSignalsRuntimeAppEmulator/ViewModels/SignalsControlViewModel.cs [hunk 2]
```diff
@@ -282,26 +361,36 @@ public string SelectedResource
...
-        private void RunSignal()
+        private async void RunSignal()
         {
-            MessageBox.Show("Comming soon...");
+            List<ResourceIdentifier> resources = new List<ResourceIdentifier>() { this.selectedResource };
+            var analysisRequest = new AnalysisRequest(resources, DateTime.UtcNow, TimeSpan.FromHours(1), this.analysisServicesFactory);
+
+            try
+            {
+                await this.smartSignal.AnalyzeResourcesAsync(analysisRequest, this.tracer, CancellationToken.None);
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/ViewModels/SignalsControlViewModel.cs]: >await this.smartSignal.AnalyzeResourcesAsync [](start = 16, length = 44)

you need to save the results

**Comment by bermanguy** [comment]
>                 <Button Style="{StaticResource RoundedButton}" HorizontalAlignment="Left" Content="Start analysis" Command="{Binding RunSignalCommand}" />

I know, Im taking care of this in next iterations

---
In reply to: [361213422](https://github.com/liadtal/Smart-Alerts/pull/55#issuecomment-361213422) [](ancestors = 361213422)

---
Refers to: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml:55 in c1e6417. [](commit_id = c1e6417df32933139bc81cdf5dba5f72e21b4f7b, deletion_comment = False)

File: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/EmulationStatusControl.xaml [hunk 3]
```diff
@@ -57,13 +57,13 @@
...
         </TextBlock>
-
+        
         <TextBlock Style="{StaticResource ControlText}" Text="Traces" Margin="0,0,0,10"/>
         <TextBox x:Name="TracerBox" Margin="0,0,0,30" Height="200" AcceptsReturn="True" 
                  Background="Transparent" BorderBrush="#FF7F7F7F" BorderThickness="1"
                  IsReadOnly="True" IsTabStop="False" HorizontalScrollBarVisibility="Visible" VerticalScrollBarVisibility="Auto"/>
 
-        <Button Style="{StaticResource RoundedButton}" HorizontalAlignment="Left" Content="Cancel"/>
+        <Button Style="{StaticResource RoundedButton}" HorizontalAlignment="Left" Content="Cancel" Margin="0,15,0,0"/>
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/Controls/EmulationStatusControl.xaml]: >Cancel [](start = 91, length = 6)

use this to cancel the analysis

File: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml [hunk 4]
```diff
@@ -58,7 +58,7 @@
 
         <Border Grid.Column="1" BorderThickness="0" Padding="20,20,20,0">
             <StackPanel Orientation="Vertical">
-                <TextBlock Style="{StaticResource SectionHeader}" Text="EMULATION STATUS"/>
+                <TextBlock Style="{StaticResource SectionHeader}" Text="STATUS"/>
                 <TextBlock x:Name="BeforeStartMessage" Text="Click 'Start analysis' to begin the emulation." Foreground="#FF767171" FontSize="16" Margin="0,80,0,0" HorizontalAlignment="Center"/>
                 <local:EmulationStatusControl x:Name="EmulationStatusControl" Visibility="Collapsed"/>
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalsControl.xaml]: please bind the visibility of these items

File: src/SDK/SmartSignalsRuntimeAppEmulator/ViewModels/SignalsControlViewModel.cs [hunk 5]
```diff
@@ -71,8 +99,8 @@ public SignalsControlViewModel(AzureResourceManagerClient azureResourceManagerCl
             this.ReadResourceTypesTask = new ObservableTask<ObservableCollection<string>>(
                 Task.FromResult(new ObservableCollection<string>()));
 
-            this.ReadResourcesTask = new ObservableTask<ObservableCollection<string>>(
-                Task.FromResult(new ObservableCollection<string>()));
+            this.ReadResourcesTask = new ObservableTask<ObservableCollection<ResourceIdentifier>>(
+                Task.FromResult(new ObservableCollection<ResourceIdentifier>()));
 
             // Initialize commands
             this.RunSignalCommand = new CommandHandler(() => this.RunSignal());
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/ViewModels/SignalsControlViewModel.cs]: remove this and use => in the property

**Pull request merged**